omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/converter/openmrs2_0/VisitConfigurationConverter2_0.java

```diff
@@ -0,0 +1,84 @@
+import org.openmrs.module.webservices.rest.web.representation.FullRepresentation;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingConverter;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.response.GenericRestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.v1_0.controller.openmrs1_9.SessionController1_9;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_9.VisitTypeResource1_9;
+import org.openmrs.module.webservices.rest.web.v1_0.wrapper.VisitConfiguration;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+@Handler(supports = VisitConfiguration.class, order = 0)
+public class VisitConfigurationConverter2_0 extends BaseDelegatingConverter<VisitConfiguration> {
+
+	private static final Logger log = LoggerFactory.getLogger(VisitConfigurationConverter2_0.class);
+
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("encounterVisitsAssignmentHandler");
+		description.addProperty("enableVisits");
+		description.addProperty("startAutoCloseVisitsTask");
+		if (rep instanceof FullRepresentation) {
+			description.addProperty("visitTypesToAutoClose");
+		}
+		return description;
+	}
+
+	@Override
+	public VisitConfiguration newInstance(String type) {
+		return new VisitConfiguration();
+	}
+
+	@Override
+	public VisitConfiguration getByUniqueId(String string) {
+		throw new ResourceDoesNotSupportOperationException();
+	}
+
+	@Override
+	public SimpleObject asRepresentation(VisitConfiguration delegate, Representation rep) throws ConversionException {
+		SimpleObject configuration = new SimpleObject();
+		configuration.add("encounterVisitsAssignmentHandler", delegate.getEncounterVisitsAssignmentHandler());
+		configuration.add("enableVisits", delegate.getEnableVisits());
+		configuration.add("startAutoCloseVisitsTask", delegate.getStartAutoCloseVisitsTask());
+
+		if (rep instanceof FullRepresentation) {
+			VisitTypeResource1_9 visitTypeResource = (VisitTypeResource1_9) Context.getService(RestService.class)
+					.getResourceBySupportedClass(VisitType.class);
+			List<SimpleObject> visitTypesToAutoClose = new ArrayList<>();
+			for (VisitType visitType : delegate.getVisitTypesToAutoClose()) {
+				try {
+					visitTypesToAutoClose.add(visitTypeResource.asDefaultRep(visitType));
+				}
+				catch (Exception e) {
+					log.error("Cannot convert VisitType to REST Resource", e);
```

Comment:
Why log when you are going to throw an exception?

Reply:
https://github.com/openmrs/openmrs-module-webservices.rest/pull/489/commits/1d775855706d43c076dcac8faf07ec2c07b7a077